### PR TITLE
Throw is not loud error

### PR DESCRIPTION
**Pull request by lucafavatella**
Before merging this PR, go over this checklist:
- [ ] The issue was approved by a considerable number of votes
- [ ] The rule is listed in the table of contents
- [ ] The rule text respects the structure:
    - [ ] Title
    - [ ] Short Description
    - [ ] Examples
    - [ ] Reasoning
- [ ] It provides links to examples in code for
    - [ ] good scenarios
    - [ ] bad scenarios

### CODE REVIEW

File: src/loud_errors.erl [hunk 1]
```diff
@@ -1,18 +1,26 @@
...
-bad(WithThis) ->
+bad1(WithThis) ->
   try
     something:that(may, fail, WithThis)
   catch
     _:Error ->
       {error, Error}
   end.
 
+bad2(WithThis) ->
```
Comment (gadgetci) [inline src/loud_errors.erl]: According to **Xref**:
> `loud_errors:bad2/1` calls undefined function `something:that/3`

File: src/loud_errors.erl [hunk 2]
```diff
@@ -1,18 +1,26 @@
...
+  try
+    something:that(may, fail, WithThis)
+  catch
+    _:Error ->
+      throw({error, Error})
+  end.
+
 good(WithThis) ->
```
Comment (gadgetci) [inline src/loud_errors.erl]: According to **Xref**:
> `loud_errors:good/1` calls undefined function `something:that/3`

File: src/loud_errors.erl [hunk 3]
```diff
@@ -1,18 +1,26 @@
 -module(loud_errors).
 
--export([bad/1, good/1]).
+-export([bad1/1, bad2/1, good/1]).
 
-bad(WithThis) ->
+bad1(WithThis) ->
```
Comment (gadgetci) [inline src/loud_errors.erl]: According to **Xref**:
> `loud_errors:bad1/1` calls undefined function `something:that/3`

File: src/loud_errors.erl (hunk 2 above)
Comment (gadgetci) [inline src/loud_errors.erl]: According to **Xref**:
> `loud_errors:good/1` calls undefined function `lager:error/2`

**Comment by lucafavatella** [comment]
I am proposing this following various untagged `throw` in sumo.erl (see [an example](https://github.com/inaka/sumo_db/blob/0.7.1/src/sumo.erl#L127)) that caused me taking a while for understanding why sumo_db_riak tests were failing locally for me.

**Pull request merged**